octoprint_mrbeam/iobeam/laserhead_handler.py

```diff
@@ -323,36 +363,49 @@ def _calculate_power_correction_factor(self):
-                    data = yaml.safe_load(stream)
-
-                if data:
-                    self._lh_cache = data.get("laser_heads")
-                    self._last_used_lh_serial = data.get("last_used_lh_serial")
-                    self._last_used_lh_model_id = data.get("last_used_lh_model")
-                    self._correction_settings = data.get("correction_settings")
-            except IOError:
-                self._logger.exception(
-                    "Can't read _laser_heads_file file: %s", self._laser_heads_file
-                )
-        else:
-            self._logger.warn("The laser_heads.yaml file can't be found at this location {}".format(self._laser_heads_file))
+        """
+        Loads laser head data from a file
+        """
+        self._logger.debug("Loading data from  {} started...!".format(self._laser_heads_file))
+        try:
+            with open(self._laser_heads_file, "r") as stream:
+                data = yaml.safe_load(stream)
+            if data:
+                self._lh_cache = data.get("laser_heads")
+                self._last_used_lh_serial = data.get("last_used_lh_serial")
+                self._last_used_lh_model = data.get("last_used_lh_model")
+                self._last_used_lh_model_id = data.get("last_used_lh_model_id")
+                self._correction_settings = data.get("correction_settings")
+
+                self._logger.info("Loading data from  {} is successful!".format(self._laser_heads_file))
+            else:
+                self._logger.warn(
+                    "Couldn't load laser head data, file: {} data: {}".format(self._laser_heads_file, data))
+        except IOError as e:
+            self._logger.error("Can't open file: {} {}".format(self._laser_heads_file, e))
+
+    def _write_laser_heads_file(self, laser_heads_file=None):
+        """
+        Overwrites the file containing the laser heads info and creates the file if it doesn't exist
 
-    def _write_laser_heads_file(self, file=None):
-        self._logger.info("Writing to laser_heads.yaml...")
+        Args:
+            laser_heads_file (Yaml): path to the file to be used
 
+        Returns:
+            None
+        """
         data = dict(
             laser_heads=self._lh_cache,
             correction_settings=self._correction_settings,
             last_used_lh_serial=self._current_used_lh_serial,
-            last_used_lh_model=self._current_used_lh_model_id,
+            last_used_lh_model=self._current_used_lh_model,
+            last_used_lh_model_id=self._current_used_lh_model_id,
         )
-        file = self._laser_heads_file if file is None else file
+        laser_heads_file = self._laser_heads_file if laser_heads_file is None else laser_heads_file
+        self._logger.info("Writing to file: {} ..started".format(laser_heads_file))
         try:
-            with open(file, "w") as outfile:
+            with open(laser_heads_file, "w+") as outfile:
```

Comment:
Why would you append to read from the file here?

Reply:
Sry, why would you want to read the file here?

Reply:
changed